### PR TITLE
Track 2 Migration for Guest Configuration

### PR DESCRIPTION
**Pull request by anineela**
# Contributing to the Azure SDK

Our latest spec is already merged to main: https://github.com/Azure/azure-rest-api-specs/tree/main/specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2022-01-25

Build succeeds:
image

Tests:
image

Currently, the "GetAllAsync" list tests are disabled due to a bug fix being done by Yao. Brought this to Arthur Ma and Yao Kou's attention.

### CODE REVIEW

**Comment by anineela** [comment]
everything is ready. Currently, blocked due to Track 2 code bug. Brought this to Arthur Ma and Yao Kou's attention.

Bug:
I noticed the SDK generator has a bug that hardcodes:

internal RestOperations class is hard coding the namespace [here](https://github.com/anineela/azure-sdk-for-net/blob/neela-update-tests-track2/sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/src/Generated/RestOperations/GuestConfigurationAssignmentsRestOperations.cs#L51). It looks like by pointing both paths to the same resource we eliminated [this operationgroup](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2022-01-25/guestconfiguration.json#L475)

Without this, all the calls for VMSS and HCRP fails.

**Comment by ArthurMa1978** [comment]
Thank you @anineela !

@Yao725  please help check the failed tests

**Comment by anineela** [comment]
@ArthurMa1978 @Yao725 I have updated the tests with the updated models (bug fix done by Yao) for the HCRP and VMSS. I am currently blocked on the List tests since that is not yet fixed (Yao informed me that). Waiting for that to be fixed or else this SDK won't be very useful. Again, everything on my end for this is done.

File: sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Azure.ResourceManager.GuestConfiguration.Tests.csproj [hunk 1]
```diff
@@ -2,4 +2,10 @@
   <ItemGroup>
     <ProjectReference Include="..\src\Azure.ResourceManager.GuestConfiguration.csproj" />
   </ItemGroup>
+  <ItemGroup>
```
Comment (Yao725) [inline sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Azure.ResourceManager.GuestConfiguration.Tests.csproj:5]: Why do we need this?

File: sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Scenario/GuestConfigurationAssignmentCollectionTests.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.GuestConfiguration.Models;
+using Azure.ResourceManager.GuestConfiguration.Tests.Utilities;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.GuestConfiguration.Tests.Scenario
+{
+    [TestFixture]
```
Comment (Yao725) [inline sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Scenario/GuestConfigurationAssignmentCollectionTests.cs:12]: Please remove this attribute and re-record all the tests. We need to make sure that both of the sync and async methods are tested.

File: sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Scenario/GuestConfigurationAssignmentCollectionTests.cs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    public class GuestConfigurationAssignmentCollectionTests : GuestConfigurationManagementTestBase
+    {
+        public GuestConfigurationAssignmentCollectionTests() : base(true)
```
Comment (Yao725) [inline sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Scenario/GuestConfigurationAssignmentCollectionTests.cs:15]:
```suggestion
        public GuestConfigurationAssignmentCollectionTests(bool isAsync) : base(isAsync)
```

File: sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Scenario/GuestConfigurationVmssAssignmentCollectionTests.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.GuestConfiguration.Tests.Utilities;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.GuestConfiguration.Tests.Scenario
+{
+    [TestFixture]
+    public class GuestConfigurationVmssAssignmentCollectionTests : GuestConfigurationManagementTestBase
+    {
+        public GuestConfigurationVmssAssignmentCollectionTests() : base(true)
```
Comment (Yao725) [inline sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Scenario/GuestConfigurationVmssAssignmentCollectionTests.cs:14]: Same as above

File: sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Scenario/GuestConfigurationHcrpAssignmentCollectionTests.cs [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+using Azure.ResourceManager.GuestConfiguration.Models;
+using Azure.ResourceManager.GuestConfiguration.Tests.Utilities;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.GuestConfiguration.Tests.Scenario
+{
+    [TestFixture]
+    public class GuestConfigurationHcrpAssignmentCollectionTests : GuestConfigurationManagementTestBase
+    {
+        public GuestConfigurationHcrpAssignmentCollectionTests() : base(true)
```
Comment (Yao725) [inline sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Scenario/GuestConfigurationHcrpAssignmentCollectionTests.cs:15]: Same as above

File: sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/tests/ScenarioTests/GuestConfigurationTests.cs [hunk 6]
```diff
@@ -20,9 +20,9 @@ public class AutomationTest
         private const string AssignmentName = "AuditSecureProtocol";
```
Comment (Yao725) [inline sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/tests/ScenarioTests/GuestConfigurationTests.cs:20]: Please revert this change, this file should not be changed.

**Review by Yao725 — COMMENTED** [review]
Please run `azure-sdk-for-net\eng\scripts\Export-API.ps1 guestconfiguration` to update the api file.

File: sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Azure.ResourceManager.GuestConfiguration.Tests.csproj (hunk 1 above)
Comment (anineela) [reply sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Azure.ResourceManager.GuestConfiguration.Tests.csproj]: removed it since it was not required.

File: sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Scenario/GuestConfigurationAssignmentCollectionTests.cs (hunk 2 above)
Comment (anineela) [reply sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Scenario/GuestConfigurationAssignmentCollectionTests.cs]: removed

File: sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Scenario/GuestConfigurationAssignmentCollectionTests.cs (hunk 3 above)
Comment (anineela) [reply sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Scenario/GuestConfigurationAssignmentCollectionTests.cs]: suggestion applied

File: sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Scenario/GuestConfigurationVmssAssignmentCollectionTests.cs (hunk 4 above)
Comment (anineela) [reply sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/tests/Scenario/GuestConfigurationVmssAssignmentCollectionTests.cs]: suggestion applied

File: sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/tests/ScenarioTests/GuestConfigurationTests.cs (hunk 6 above)
Comment (anineela) [reply sdk/guestconfiguration/Microsoft.Azure.Management.GuestConfiguration/tests/ScenarioTests/GuestConfigurationTests.cs]: reverted

**Comment by anineela** [comment]
> Please run `azure-sdk-for-net\eng\scripts\Export-API.ps1 guestconfiguration` to update the api file.

**Comment by anineela** [comment]
All code review comments have been resolved.

**Review by ArthurMa1978 — APPROVED** [review]
